main.go

```diff
@@ -48,6 +49,7 @@ var (
 
 	dealsListFields = []string{"ID", "DealSize", "DealDuration", "Status", "AskPrice", "VerifiedAskPrice",
 		"StartEpoch", "DataURIFetchAttempts", "CreatedAt", "ErrorCause"}
+	validStorageProviderID = regexp.MustCompile("^[a-z]0[0-9]+$")
```

Comment:
Actually there are only "f0..." and "t0..." https://docs.filecoin.io/about-filecoin/how-filecoin-works/#id-addresses-f0, but just don't put too rigid rules